activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,10 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        !mariadb? && version >= "5.7.5"
```

Comment:
@kamipo

Thanks for the suggestion. I still prefer my idea but I do not have strong opinion to handle this 
by adding comments at the `supports_virtual_columns?` then I'd like to hear from Jeremy.

@jeremy

I'd like to get some opinion from MariaDB users (I think you are the one of them) 
are happy to get somewhat limited virtual column support - migration is supported but schema dumper is not supported. Or prefer to wait until MariaDB information_schema can generate virtual column expressions like MySQL 5.7 does.

Thanks guys.